### PR TITLE
feat(ReleaseUi): Add other / detected license in release

### PR DESCRIPTION
**Pull request by akapti**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
 * PR to display other license from CLI file and add it to `other license` attribute of `release`.
 * Which issue is this pull request belonging to and how is it solving it? #1051 
 * Did you add or update any new dependencies that are required for your change? - `No`

Issue: #1051 

 * Attach CLI files with Main/Other license ids to a release.
 * Go to `Clearing Details` tab in `Release Details` page, and click on `Show License Info` button in `SPDX Attachments` table.
 * Clicking on `Show License Info` button should load all the licenses from the CLI.
 * Click on `Add data to this release` button to add the Main/License ids to the release.
> Have you implemented any additional tests? - `No`

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: Abdul Kapti <abdul.kapti@siemens-healthineers.com>

### CODE REVIEW

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java [hunk 1]
```diff
@@ -587,10 +589,26 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
                     }
                 });
                 jsonGenerator.writeEndArray();
-            } else if (licenseWithText != null) {
+            } else if (mainLicenseNames.size() > 0) {
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:592]: `CommonUtils.isNotEmpty(mainLicenseNames)` would look good here.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java [hunk 2]
```diff
@@ -587,10 +589,26 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
...
                 jsonGenerator.writeStringField(LICENSE_PREFIX, LanguageUtil.get(resourceBundle,"main.license.id"));
-                jsonGenerator.writeArrayFieldStart("licenseIds");
-                jsonGenerator.writeString(licenseWithText.getLicenseSpdxId());
+                jsonGenerator.writeArrayFieldStart(LICENSE_IDS);
+                mainLicenseNames.forEach(license -> {
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:595]: May be `wrapException(_)` can be used here.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java [hunk 3]
```diff
@@ -587,10 +589,26 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
...
+                        jsonGenerator.writeString(license);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+                jsonGenerator.writeEndArray();
+                jsonGenerator.writeStringField("otherLicense", LanguageUtil.get(resourceBundle,"other.license.id"));
+                jsonGenerator.writeArrayFieldStart("otherLicenseIds");
+                otherLicenseNames.forEach(license -> {
+                    try {
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:606]: `wrapException(_)` here too?

**Review by smrutis1 — CHANGES_REQUESTED** [review]
few minor comments.

**Comment by smrutis1** [comment]
Tested it as per the mentioned steps successfully. However observed one issue(Seems to be there in current master too), clicking on the added license on release summary page redirects to component summary page.

**Comment by smrutis1** [comment]
Some UI issues while editing.
image

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java (hunk 1 above)
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java]: added the requested change.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java (hunk 2 above)
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java]: done

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java (hunk 3 above)
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java]: done.

**Comment by akapti** [comment]
> Some UI issues while editing.

Fixed UI issues and incorporated the code review changes.

**Review by smrutis1 — APPROVED** [review]
Code looks good.

**Comment by akapti** [comment]
From Discussion: Removed License link from `Component` and `Release` details page.

**Comment by JaideepPalit** [comment]
Seems to be some issues. - 
- License Link in Project Clearing status, seems to be redirecting to Project Details Page.
- For some spdx file (shared separately), getting below error on `Add data to release`
image

> From Discussion: Removed License link from `Component` and `Release` details page.
 - working fine

**Comment by akapti** [comment]
> * License Link in Project Clearing status, seems to be redirecting to Project Details Page.

Removed License link for all the  pages where wrong link was being generated.

> * For some spdx file (shared separately), getting below error on `Add data to release`

Fixed the issue.

**Review by JaideepPalit — APPROVED** [review]
Tested successfully

**Pull request merged**